File: BOJ/[1083] 소트/KingDonggyu/1083.js
function solution(N, A, S) {
  for (let x = 0; x < N; x++) {
    const max = { index: x, value: A[x] };

    for (let y = x + 1; y <= x + S; y++) {
      if (y >= N) break;

      if (max.value < A[y]) {
        max.index = y;
        max.value = A[y];
      }
    }

    for (let i = max.index; i > x; i--) {
      [A[i], A[i - 1]] = [A[i - 1], A[i]];
      S--;
    }

    if (!S) break;
  }

  return A.join(' ');
}

const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

const input = [];

rl.on('line', (line) => {
  input.push(line);
  if (input.length === 3) {
    rl.close();
  }
}).on('close', () => {
  console.log(
    solution(
      +input[0],
      input[1].split(' ').map((ch) => +ch),
      +input[2]
    )
  );
  process.exit();
});

File: BOJ/[1083] 소트/Johoseong/1083.py
import sys
input = sys.stdin.readline

N = int(input())
arr = list(map(int, input().split(" ")))
S = int(input())

for i in range(N - 1):
    if S == 0:
        break
    tmp_max = arr[i]
    index = i
    for j in range(i + 1, min(N, i + 1 + S)):
        if tmp_max < arr[j]:
            tmp_max = arr[j]
            index = j
    S -= index - i
    for j in range(index, i, -1):
        arr[j] = arr[j - 1]
    arr[i] = tmp_max
    
print(*arr)